### PR TITLE
[AIP] Add draft AIP for consensus observer.

### PR DESCRIPTION
**Pull request by JoshLind**
This PR adds a draft AIP for consensus observer.

### CODE REVIEW

File: aips/aip-xx.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+There are several risks and drawbacks associated with CO, including:
+1. **Transaction Re-execution**: One of the benefits of state sync is that it provides an
+   [intelligent syncing](https://aptos.dev/en/network/nodes/configure/state-sync#intelligent-syncing)
+   mode, where fullnodes can skip transaction re-execution and reuse the execution results from validators.
+   This helps to reduce CPU usage and improve synchronization time. However, CO does not
+   provide this as it requires fullnodes to re-execute blocks in real-time. As a result, it requires all fullnodes
+   to have equivalent (or better) hardware resources than validators, to ensure they can process blocks at the
+   same rates while dealing with other overheads (e.g., REST API requests).
+      - **Mitigation**: While this is a drawback, it does not pose a significant risk, as fullnodes are already
```
Comment (sherry-x) [inline aips/aip-xx.md:154]: Can a fullnode choose to not use CO, accept the latency penalty at high load, and potentially save hardware cost if they want to? or is this going to be an enforced sync mode for all nodes?

File: aips/aip-xx.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+      required to deploy equivalent hardware resources as validators. Moreover, the benefits of CO in terms of
+      reduced latencies are more than sufficient to outweigh the additional costs of transaction re-execution.
+2. **CO Complexity**: CO will introduce additional complexity to the Aptos node software, as new components
```
Comment (sherry-x) [inline aips/aip-xx.md:157]: Is there any extra bandwidth cost because of the new network messages?

File: aips/aip-xx.md (hunk 1 above)
Comment (JoshLind) [reply aips/aip-xx.md:154]: Yes, it's config gated, so nodes can opt-out and use regular state sync if they want. I'll update the text 😄

File: aips/aip-xx.md (hunk 2 above)
Comment (JoshLind) [reply aips/aip-xx.md:157]: No, if anything it'll be less data because nodes will be syncing transactions instead of outputs. 😄

**Review by sherry-x — APPROVED** [review]

**Pull request merged**